pkg/registries/azure/azure.go

```diff
@@ -22,7 +58,99 @@ func CreatorWithoutRepoList() (string, types.Creator) {
 	return types.AzureType,
 		func(integration *storage.ImageIntegration, options ...types.CreatorOption) (types.Registry, error) {
 			cfg := types.ApplyCreatorOptions(options...)
-			reg, err := docker.NewDockerRegistry(integration, true, cfg.GetMetricsHandler())
+			reg, err := newRegistry(integration, true, cfg.GetMetricsHandler())
 			return reg, err
 		}
 }
+
+func validate(cfg *storage.AzureConfig) error {
+	errorList := errorhelpers.NewErrorList("azure container registry validation")
+	if cfg.GetEndpoint() == "" {
+		errorList.AddString("endpoint must be specified for Azure container registry (e.g. <registry>.azurecr.io)")
+	}
+	if !cfg.GetWifEnabled() && cfg.GetPassword() == "" {
```

Comment:
if WIF is enabled and someone sets a password, is it just ignored? Similarly, is that allowed?

Reply:
Yes if WIF is enabled, the password will be ignored. It therefore does not make sense to set a password with WIF, but I'm not sure we should actually fail the validation in this case. What do you think?

Reply:
sounds good to me